understanding-java/README.md

```diff
@@ -0,0 +1,67 @@
+# Understanding Java
+
+No one involved in this workshop needs to know Java. The goal of this workshop is to work with code _even though we don't know the language_.
+
+## Learning objectives
+
+```objectives
+- [ ] Make sense of code in a language you don't know.
+- [ ] Identify similarities and differences between programming languages.
+- [ ] Use documentation, debugging, and search techniques to fix a bug in a program in a language you don't know.
+- [ ] Use documentation, debugging, and search techniques to extend a program in a language you don't know.
+```
+
+## Introduction
+
+There are a lot of things in common across different programming languages.
+
+Even if you don't know Java, knowing other languages gives you a basis for making sense of it.
+
+All of your existing documentation/research/Googling skills will help you!
+
+A sample complete implementation of this exercise can be found in https://github.com/CodeYourFuture/CYF-Workshops/blob/main/understanding-java/DnsResolver.java but we recommend you try to completely solve the exercises yourself rather than looking at the sample implementation.
+
+## Understanding code
+
+Here is some Java code:
+
+```java
+class DnsResolver {
+    public static void main(String[] args) {
+        if (args.length != 1) {
+            System.err.printf("Expected exactly one argument, but got %d%n", args.length);
+            System.exit(1);
+        }
+        String host = args[0];
+
+        String[] parts = host.split(".");
+        System.out.printf("The top-level domain of the host %s is %s%n", host, parts[parts.length - 1]);
+    }
+}
+```
+
+Take 10 minutes to play computer with this code. Read through it line-by-line and work out what you think each line does.
```

Comment:
Suggest being more directive -- is this a group workshop? Should they be working in pairs on this code? Whole group? Who should be reading through?